### PR TITLE
Delete folder

### LINKED ISSUE #391: Don't create destination directory if there's no data to download / upload

### PR DESCRIPTION
**Pull request by zishanmirza**
This patch deletes the folder before "DownloadError" and "UploadError".

Fixes #391.

Before submitting a PR to the `dev` branch:
- [X] Tests passing
- [X] Black formatting
- [X] Rebase/merge the `dev` branch
- [x] Note in the CHANGELOG

### CODE REVIEW

File: dds_cli/base.py [hunk 1]
```diff
@@ -55,13 +55,15 @@ def __init__(
         force_renew_token: bool = False,
         no_prompt: bool = False,
         token_path: str = None,
+        temporary_folder: str = None,
```
Comment (inaod568) [inline dds_cli/base.py:58]: You don't need this, you just need row 86
```suggestion
```

File: dds_cli/base.py [hunk 2]
```diff
@@ -55,13 +55,15 @@ def __init__(
...
     ):
         """Initialize Base class for authenticating the user and preparing for DDS action."""
         self.project = project
         self.method_check = method_check
         self.method = method
         self.no_prompt = no_prompt
         self.token_path = token_path
+        self.temporary_folder = temporary_folder
```
Comment (inaod568) [inline dds_cli/base.py:66]:
```suggestion
```

File: dds_cli/data_getter.py [hunk 3]
```diff
@@ -99,6 +99,9 @@ def __init__(
                 )
 
             if not self.filehandler.data:
+                if self.temporary_folder and self.temporary_folder.is_dir():
+                    LOG.debug(f"Deleting folder {self.temporary_folder}.")
```
Comment (inaod568) [inline dds_cli/data_getter.py:103]:
```suggestion
                    LOG.debug(f"Deleting temporary folder {self.temporary_folder}.")
```

File: dds_cli/data_putter.py [hunk 4]
```diff
@@ -248,6 +248,9 @@ def __init__(
             progress.remove_task(wait_task)
 
         if not self.filehandler.data:
+            if self.temporary_folder and self.temporary_folder.is_dir():
+                LOG.debug(f"Deleting folder {self.temporary_folder}.")
```
Comment (inaod568) [inline dds_cli/data_putter.py:252]:
```suggestion
                LOG.debug(f"Deleting temporary folder {self.temporary_folder}.")
```

**Review by inaod568 — CHANGES_REQUESTED** [review]
I think it looks good and it works as I'd expect. Just a few changes though, there are a couple of lines you don't need

**Comment by inaod568** [comment]
@zishanmirza Regarding the new module in the `requirements.txt`. As you mentioned in the chat we should have a `requirements-test.txt` file - we should have the same structure in this repo as in the `dds_web`.

1. Create add a new file `requirements-test.txt` within the `tests/` folder 
2. Move `pytest` and `pyfakefs` from the `requirements.txt` to the `tests/requirements-test.txt`
3. Change this line to `pip install -r tests/requirements-test.txt .` (I believe this should work but try it.):
https://github.com/zishanmirza/dds_cli/blob/c573873081edeeec315d052d67ec80845bb0215c/.github/workflows/python-app.yml#L24

**Comment by zishanmirza** [comment]
Moved `pytest` and `pyfakefs` from `requirements.txt` to `tests/requirements-test.txt` and updated `python-app.yml`.

**Comment by zishanmirza** [comment]
I have moved `pytest` and `pyfakefs` from `requirements.txt` to `tests/requirements-test.txt` and updated `python-app.yml`.

**Comment by zishanmirza** [comment]
I have moved `pytest` and `pyfakefs` from `requirements.txt` to `tests/requirements-test.txt` and updated `python-app.yml`.

**Review by inaod568 — APPROVED** [review]

**Comment by inaod568** [comment]
Looks good!

**Pull request merged**